File: prog2/2. hét/Liskov.java
public class Liskov {
    public static class Program{
        public static void fgv(Negyszog negyszog) {
            negyszog.kerulet();
     }};
    
    public static class Negyszog {
        void kerulet() {}; //négyszögek kerülete=2*(a+b), 
                           //nem statikus, mert a kerület függvény nem tartozik szorosan a Négyszög osztályhoz
    };

    public static class Trapez extends Negyszog
    {};

    public static class Negyzet extends Negyszog // négyzet kerülete esetén nincs b, csak 4 a
    {};
    
    public static void main(String[] args) {
     //Program program;
     Negyszog negyszog = new Negyszog();
     Program.fgv(negyszog);

     Trapez trapez = new Trapez();
     Program.fgv(trapez);

     Negyzet negyzet = new Negyzet();
     Program.fgv(negyzet); // sérül az LSP, mert a négyzet kerületét nem számíthatjuk ki úgy, 
                              // mint a téglalapét, mert mások az adatok
    }
};

File: prog2/2. hét/Liskov.cpp
class Negyszog {
public:
     virtual void kerulet() {}; //négyszögek kerülete=2*(a+b)
};

class Program {
public:
     void fgv ( Negyszog &negyszog ) {
          negyszog.kerulet();
     }};

class Trapez : public Negyszog
{};

class Negyzet : public Negyszog // négyzet kerülete esetén nincs b, csak 4 a
{};

int main() {
     Program program;
     Negyszog negyszog;
     program.fgv ( negyszog );

     Trapez trapez;     //hacsak nem speciális eset 4 különböző hosszú oldala van
     program.fgv ( trapez );

     Negyzet negyzet;
     program.fgv ( negyzet ); // sérül az LSP, mert a négyzet kerületét nem számíthatjuk ki úgy, 
                              // mint a téglalapét, mert mások az adatok
     return 0;
}